### PR TITLE
python3: use compatible urllib packages

### PR DESCRIPTION
**Pull request by kshtsk**
Use py2/py3 compatible functions url* packages.

Python3
    from urllib.parse import parse_qs, urljoin, urlparse, urlencode
    from urllib.request import urlopen, Request
    from urllib.error import HTTPError
Python2
    from urlparse import parse_qs, urljoin, urlparse
    from urllib import urlencode
    from urllib2 import urlopen, Request, HTTPError

Signed-off-by: Kyr Shatskyy <kyrylo.shatskyy@suse.com>

### CODE REVIEW

**Comment by kshtsk** [comment]
@susebot run deploy

File: teuthology/misc.py [hunk 1]
```diff
@@ -13,13 +13,19 @@
...
 import re
 import pprint
 
+try:
+    from urllib.parse import urljoin
+    from urllib.request import urlopen
+    from urllib.error import HTTPError
+except ImportError:
+    from urlparse import urljoin
+    from urllib2 import urlopen,  HTTPError
```
Comment (smithfarm) [inline teuthology/misc.py]: `s/urlopen,  HTTPError/urlopen, HTTPError/` to remove excess white space

File: teuthology/lock/query.py [hunk 2]
```diff
@@ -1,6 +1,10 @@
 import logging
 import os
-import urllib
+
+try:
```
Comment (toabctl) [inline teuthology/lock/query.py]: We could use `six` which is already in requirements.txt to have less code. And later it's easier to grep for it to remove py2 support. Or is six not wanted?

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by kshtsk** [comment]
@susebot run deploy

File: teuthology/lock/query.py (hunk 2 above)
Comment (smithfarm) [reply teuthology/lock/query.py]: I would prefer to avoid `six`, if only because it's yet another thing I would have to learn how to use.

**Pull request merged**